src/Npgsql/Internal/PgSerializerOptions.cs

```diff
@@ -84,23 +84,20 @@ public static bool IsWellKnownTextType(Type type)
             ? ((TypeInfoCache<DataTypeName>)(_typeInfoCache ??= new TypeInfoCache<DataTypeName>(this))).GetOrAddInfo(type, pgTypeId?.DataTypeName, defaultTypeFallback)
             : ((TypeInfoCache<Oid>)(_typeInfoCache ??= new TypeInfoCache<Oid>(this))).GetOrAddInfo(type, pgTypeId?.Oid, defaultTypeFallback);
 
-    public PgTypeInfo? GetDefaultTypeInfo(PostgresType pgType)
-        => GetTypeInfoCore(null, ToCanonicalTypeId(pgType), false);
+    internal PgTypeInfo? GetTypeInfoInternal(Type? type, PgTypeId? pgTypeId)
+        => GetTypeInfoCore(type, pgTypeId, false);
 
-    public PgTypeInfo? GetDefaultTypeInfo(PgTypeId pgTypeId)
-        => GetTypeInfoCore(null, pgTypeId, false);
+    internal PgTypeInfo? GetObjectOrDefaultTypeInfo(PgTypeId pgTypeId)
+        => GetTypeInfoCore(typeof(object), GetCanonicalTypeId(pgTypeId), true);
```

Comment:
Just to check, do we really need to call `GetCanonicalTypeId` here? From my understanding that's just going to result in additional call to `DatabaseInfo` which is going to do nothing since we already call `ToCanonicalTypeId` on `PostgresType` beforehand.

Reply:
If it's already mapped it's essentially free.

Reply:
> If it's already mapped it's essentially free.

Yeah, we do check whether it's the thing we need at which point we just shortcut (unless `validate` is specified). But still, given that `GetTypeInfoInternal` doesn't do that, we might want to align with it?